### PR TITLE
Resizable sidebar

### PR DESCRIPTION
**Pull request by fschmenger**
This pull request adds additional options to the sidebar configuration to make it resizable on behalf of the user - a resizer appears when hovering the right edge of the sidebar.

Enable the option in app.conf (defaults to false):
```
"sidebar": {
    "resizable": true
}
```

Set additional resizing constraints (no restrictions per default)
```
"sidebar": {
    "minWidth": 400,
    "maxWidth": 600
}
```

A technical note:
To resize the OL map canvas automatically when it`s container size changes, I hooked up a [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) in Map component. This should have sufficient browser support by now. It also fires initially so the workaround on application start has been removed.
This also means we can remove other map-resize related code like in attributeTable. I will provide those with a couple of fixes in an upcoming PR.

Merry christmas to all of you!

### CODE REVIEW

File: src/components/ol/Map.vue
Comment (JakobMiksch) [inline src/components/ol/Map.vue:70]:
```suggestion
    // therefore flag the initialization to prevent errors.
```

File: app-starter/components/AppSidebar.vue
Comment (JakobMiksch) [inline app-starter/components/AppSidebar.vue:86]: can we use a arrow function here instead?
then we do not need to have the `me` variable, because `this` can be used in the function directly

**Review by JakobMiksch — APPROVED** [review]
Works for me and code looks fine. 
One minor note regarding formatting and a question inline. Otherwise fine to merge.
Thx for the contribution and happy new year :-)

File: app-starter/components/AppSidebar.vue
Comment (fschmenger) [reply app-starter/components/AppSidebar.vue:86]: To my knowledge declaring an arrow function and inlining it to the event handler only works, if the event doesnt have to be unregistered. So I could do it for `onMouseUp ` but not for `onMouseMove`. For clarity I decided to declare both handlers explicitly. 
I can give it a try though.

**Comment by fschmenger** [comment]
Thanks for the review and a happy new year Jakob!

File: app-starter/components/AppSidebar.vue
Comment (fschmenger) [reply app-starter/components/AppSidebar.vue]: Ok my bad. Obviously it works as well with arrow functions. You just can't inline them. Want to give it a quick test/ approval again before I merge?

**Review by JakobMiksch — APPROVED** [review]
@fschmenger the change in https://github.com/wegue-oss/wegue/pull/313/commits/a9d7a5c3b4b73af330e134be38e9cb73dba2b2eb looks fine to me. Please merge at will.

**Pull request merged**
